### PR TITLE
[KEYCLOAK-9922] Migrate from dep to Go 1.11 modules

### PR DESCRIPTION
**Pull request by abstractj**

### CODE REVIEW

**Comment by abstractj** [comment]
@fredbi would you like to review?

File: Makefile [hunk 1]
```diff
@@ -27,14 +27,14 @@ build: golang
 	@mkdir -p bin
 	go build -ldflags "${LFLAGS}" -o bin/${NAME}
 
-release: clean golang deps
```
Comment (fredbi) [inline Makefile]: I assume some other PR maintains the Dockerfile at https://github.com/jboss-dockerfiles/keycloak/tree/master/gatekeeper (same for k8s charts)

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,11 @@
 language: go
+
 go:
-- "1.11.x"
+- 1.11.x
```
[the remark this answers is not in the document]
Comment (abstractj) [reply .travis.yml]: @fredbi the latest release is 1.12.x, but the standard versions distributed in some distros like Fedora is 1.11.x. We can provide a matrix for testing 1.12.x, but still we need to test 1.11.x.

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+require (
+	github.com/PuerkitoBio/purell v1.1.0
+	github.com/armon/go-proxyproto v0.0.0-20180202201750-5b7edb60ff5f
+	github.com/boltdb/bolt v1.3.1
+	github.com/coreos/go-oidc v0.0.0-20171020180921-e860bd55bfa7
+	github.com/elazarl/goproxy v0.0.0-20181111060418-2ce16c963a8a
+	github.com/fsnotify/fsnotify v1.4.7
+	github.com/go-chi/chi v3.3.3+incompatible
+	github.com/pressly/chi v3.3.3+incompatible
```
[the remark this answers is not in the document]
Comment (abstractj) [reply go.mod]: @fredbi thanks, updated.

File: Makefile (hunk 1 above)
Comment (abstractj) [reply Makefile]: @fredbi not sure if I follow you on this. Could you please elaborate?

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (fredbi) [reply .travis.yml]: agree. I meant:
```yaml
- 1.11.x
- 1.12.x
```

File: Makefile (hunk 1 above)
Comment (fredbi) [reply Makefile]: oh you're right. I've seen that the Dockerfile there is building from binary release, not source.
My bad. It is constrasting with our own Dockerfiles which build most things from source...

**Comment by fredbi** [comment]
Just the .travis.yml to explictly state we are supporting 2 latest go relases and that's it! Thanks

File: Makefile (hunk 1 above)
Comment (abstractj) [reply Makefile]: Absolutely no worries, thanks for reviewing it.

**Pull request merged**